### PR TITLE
Debug logger is able to be overriden

### PR DESCRIPTION
**Pull request by Dmitry-Ostashev**
This PR is needed for https://github.com/DevExpress/testcafe-studio/issues/2729 to hide console output in coded steps with debug action.

### CODE REVIEW

**Comment by Dmitry-Ostashev** [comment]
@testcafe-build-bot retest

File: src/runner/index.js [hunk 1]
```diff
@@ -183,6 +184,16 @@ export default class Runner extends EventEmitter {
         assets.forEach(asset => this.proxy.GET(asset.path, asset.info));
     }
 
+    _validateDebugLogger () {
+        const debugLogger = this.configuration.getOption(OPTION_NAMES.debugLogger);
+
+        if (debugLogger === void 0) {
+            this.configuration.mergeOptions({
```
Comment (helen-dikareva) [inline src/runner/index.js]: strictly speaking we need to check that a custom `debugLogger` has `showBreakpoint` and `hideBreakpoint` methods

File: test/server/runner-test.js [hunk 2]
```diff
@@ -759,6 +759,23 @@ describe('Runner', () => {
                     expect(consoleWrapper.messages.log).eql(null);
                 });
         });
+
+        it('Should use default debugLogger if need', () => {
```
Comment (helen-dikareva) [inline test/server/runner-test.js]: all texts in open source repo have to be checked

File: src/runner/index.js [hunk 3]
```diff
@@ -187,7 +187,10 @@ export default class Runner extends EventEmitter {
     _validateDebugLogger () {
         const debugLogger = this.configuration.getOption(OPTION_NAMES.debugLogger);
 
-        if (debugLogger === void 0) {
+        const debugLoggerDefinedCorrectly = debugLogger && 
+            ["showBreakpoint", "hideBreakpoint"].every(method => method in debugLogger && debugLogger[method] instanceof Function);
```
Comment (helen-dikareva) [inline src/runner/index.js]: we can use lodash `isFunction`

**Pull request merged**